Site Creation: Site Info page improvements

oguzkocer opened the pull request:
Fixes #8736 & #8766. This PR contains a set of small improvements mostly to the site info screen as well as fixing our master branch's build which was broken due to merging `develop` into it. Here is a list of changes:

* We now request focus for the title text field as soon as the user enters site info page - technically, there is a `500ms` delay. It also adds a unit test for it.
* Fixes an issue in the segments page where an empty icon color was causing a crash. It also adds a `TODO` so we can check in with the designers on how best to handle this issue. a413760fa4d2498437f80864dff101f3b5c7a0a7
* Fixes the site info and vertical view model tests due to changes in FluxC
* Updates the site info page layout to keep the `skip/next` button visible at all times as described in #8736. It looks like the layout will be scrollable in most cases, so the shadow is visible at all times. This is consistent with the new domains page. We'll address any concerns about it during the final design review if necessary.
* Updates the `Optional Tagline` text to `Tagline` in site info page as described in #8766.

We'll be doing a full design review at the end of site creation project, so a design review is not required for this PR.

To test:
* Go to site info screen (check the gif below to see how to access it)
* Check that the `skip/next` button and its container is fixed to bottom and the content above it is scrollable
* Check that the `title` text field is selected and the keyboard shows up as you enter the screen.
* Feel free to run the unit tests that were changed added, but Travis already does it for us, so it's not necessary.

Here is a gif that's showcasing the changes and demonstrates how to access the site info screen: https://cloudup.com/cyPiVq-Zt2b

One reviewer should be enough for this PR, I only asked two developers so whoever is available can take it.

_P.S: I recommend turning on ` Hide whitespace changes` in the `Diff settings` while reviewing this PR as it'll make it much easier to see the actual layout changes._

oguzkocer commented on WordPress/src/main/java/org/wordpress/android/ui/sitecreation/NewSiteCreationSiteInfoFragment.kt:
```diff
@@ -63,7 +64,6 @@ class NewSiteCreationSiteInfoFragment : NewSiteCreationBaseFormFragment<NewSiteC
     }
 
     override fun setupContent(rootView: ViewGroup) {
-        // TODO: Get the title from the main VM
```
This was handled in a previous PR but we forgot to remove the `TODO` for it.

oguzkocer commented on WordPress/src/main/java/org/wordpress/android/ui/sitecreation/segments/NewSiteCreationSegmentViewHolder.kt:
```diff
@@ -35,8 +35,10 @@ sealed class NewSiteCreationSegmentViewHolder(internal val parent: ViewGroup, @L
+                icon.setColorFilter(Color.parseColor(iconColor), PorterDuff.Mode.SRC_IN)
             } catch (e: IllegalArgumentException) {
                 AppLog.e(AppLog.T.SITE_CREATION, "Error parsing segment icon color: ${uiState.iconColor}")
             }
```
> The R.color.black.toString() statement converts the black color resource ID int into a String setting the iconColor value to something like "2131099691" rather than getting the text value of the black color reference like "#ff000000" as expected.

I should have checked that, nice catch.

> Rather than setting the default iconColor value programmatically, what about adding the android:tint attribute to the icon view in the new_site_creation_segment_item layout and updating the color filter programmatically if the uiState.iconColor is not blank?

Unfortunately that won't work since this is a `ViewHolder` if it's being reused, it could have the color from the previous use instead of the one set in the layout.

I will fix this with a temporary black color for now and will do a proper fix once we know what we actually want to have happen. We might treat the missing `iconUrl` and `iconColor` as invalid states and there is still some discussion going on about this with our backend friends.

theck13 replied:
> Unfortunately that won't work since this is a `ViewHolder` if it's being reused, it could have the color from the previous use instead of the one set in the layout.

Ahh, you're right.  It seems like the only solution is to set it programmatically then.

> I will fix this with a temporary black color for now and will do a proper fix once we know what we actually want to have happen. We might treat the missing `iconUrl` and `iconColor` as invalid states and there is still some discussion going on about this with our backend friends.

Sounds good.

oguzkocer replied:
Added the temporary fix: 5553b16c4b7dae6408825f6b873f7c1eb2b12c4c

oguzkocer commented:
> The subtitle in the error with retry layout, subtitle in the 1 of 3 screen, and tagline description in the 2 of 3 screen all have widows. We may want to update those views to WPTextView and add the app:fixWidowWords="true" attribute to avoid that. See the screenshots below for illustration.

@theck13 Are you sure that we need to handle this with a custom solution? I am not sure how I feel about this hacky solution we have (which I only just learned about), but more importantly I can't seem to be able to reproduce the widow words at all. The different resolution or the different API might of course be playing a role here, but I have been trying to update the string to get that to happen and it looks like it's handled properly in all cases in my device. Are you able to reproduce this with a specific device / API combination I could test?

theck13 commented:
> Are you able to reproduce this with a specific device / API combination I could test?

I used the stock emulator using the Google Pixel 2 API 25 image to take the screenshots shown below.

![site_creation_widows](https://user-images.githubusercontent.com/3827611/50424475-796bb680-0832-11e9-8873-5da54ac65c1b.png)

oguzkocer commented:
Thanks for the detailed review @theck13! I've addressed the black color issue in 5553b16c4b7dae6408825f6b873f7c1eb2b12c4c and addressed the punctuation and toolbar issues in fba0089f820ab71a242afbcfcd01ea6f2b7a0b34.

For the widow word issue, I have been trying to download API 25 to reproduce but for some reason it's been taking forever and I ended up canceling it. I have added a note for this in the design issues master issue, so I can follow up with it later. #8864 

It should be ready for another round.

theck13 commented on WordPress/src/main/java/org/wordpress/android/ui/sitecreation/segments/NewSiteCreationSegmentViewHolder.kt:
```diff
@@ -35,8 +35,10 @@ sealed class NewSiteCreationSegmentViewHolder(internal val parent: ViewGroup, @L
             title.text = uiState.title
             subtitle.text = uiState.subtitle
             imageManager.load(icon, IMAGE, uiState.iconUrl)
```
Should we add a case to use a default icon when `iconUrl` is empty similar to `iconColor`?  The fix for `iconColor` below works, but the icon is still not visible as shown below since the `iconUrl` is empty also.

![site_creation_segments_icon](https://user-images.githubusercontent.com/3827611/50411509-7dfe8380-07ce-11e9-855c-c40f942a8db9.png)

I see one of the items in https://github.com/wordpress-mobile/WordPress-Android/issues/8864  is:
> What should we do if the `iconUrl` or `iconColor` is missing for a segment? This is currently the case for the `Online Store` segment. We have a `TODO` for this in `NewSiteCreationSegmentViewHolder` as we are just setting the color to black if it's missing, but don't do anything about the missing `iconUrl`.

Just let me know if we're deferring the default `iconUrl` value until getting input from designers.

oguzkocer replied:
We are just leaving it as is since we have an issue for it. It's really not worth the effort right now, it's a very simple fix once we know what we want to do with it. Thanks for checking in again, sorry if it wasn't very clear.

P.S: Keep in mind that this is merging to a WIP branch and it's also behind a feature flag, so as long as we have an issue we are completely fine merging non-polished PRs.
